snake_state.py

```diff
@@ -136,6 +137,7 @@ def __init__(self, size, seed):
         self.__snake = deque([self.__grid.new_cell(self.__grid.find_free_cell(), GridCellType.SNAKE_HEAD)])
         self.__alive = True
         self.__apple = self.__grid.new_cell(self.__grid.find_free_cell(), GridCellType.APPLE)
+        self.__dist_to_apple = self._get_dist_to_apple(self.__snake[0])
```

Comment:
No need to save it, just use a dynamic property that calculates the distance when it's used